libsolidity/codegen/ir/IRGeneratorForStatements.cpp

```diff
@@ -2522,7 +2522,7 @@ void IRGeneratorForStatements::appendExternalFunctionCall(
 		if <success> {
 			<?dynamicReturnSize>
 				// copy dynamic return data out
-				returndatacopy(<pos>, 0, returndatasize())
+				returndatacopy(<pos>, 0, <returnSize>)
```

Comment:
This part is only executed in the dynamic case, so you do not need to change anything here.